src/Sulu/Bundle/AdminBundle/Resources/js/containers/FormOverlay/FormOverlay.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+import React from 'react';
+import {action, computed, observable} from 'mobx';
+import type {ElementRef} from 'react';
+import {observer} from 'mobx-react';
+import Overlay from '../../components/Overlay';
+import {translate} from '../../utils';
+import Snackbar from '../../components/Snackbar';
+import Loader from '../../components/Loader';
+import Form, {ResourceFormStore} from '../Form';
+import type {FormStoreInterface} from '../Form/types';
+import type {Size} from '../../components/Overlay/types';
+import formOverlayStyles from './formOverlay.scss';
+
+type Props = {|
+    confirmDisabled: boolean,
+    confirmLoading: boolean,
+    confirmText: string,
+    formStore?: ?FormStoreInterface,
```

Comment:
Can this really be null? Should then not a normal overlay used instead if there is no formStore?

Reply:
just to make rendering easier 🙂  i think in most cases the component will be used like this:
```
<FormOverlay
    formStore={this.formStore}
    open={!!this.formStore}
/>
```

if we make the prop non-nullable, you need to render it like this:
```
{this.formStore && (
    <FormOverlay
        formStore={this.formStore}
        open={!!this.formStore}
    />
)}
```

but dont have a too strong opinion. we can make it non-nullable too

Reply:
What happens if the FormOverlay is open and has no formStore, would it then not just render nothing. Should we not avoid incompatible property combinations? I would more go making it required, did look at the `ListOverlay` there the [`ListStore`](https://github.com/sulu/sulu/blob/3dd8b83bccf488196dcd526ede19dda0e4b52932/src/Sulu/Bundle/AdminBundle/Resources/js/containers/ListOverlay/ListOverlay.js#L22) is also required.